CMakeLists.txt

```diff
@@ -12,20 +14,25 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(cv_bridge REQUIRED)
-find_package(darknet_vendor REQUIRED)
+find_package(Darknet REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rclcpp_components REQUIRED)
 find_package(sensor_msgs REQUIRED)
 find_package(vision_msgs REQUIRED)
 
+include_directories(
+  ${Darknet_INCLUDE_DIR}
+)
+link_libraries(Darknet::dark)
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```

Comment:
What happens if this is not set?

Reply:
If `CMAKE_INSTALL_RPATH_USE_LINK_PATH` is not set this will result in a **runtime error** `[detector_node-1] /ros2_ws/install/openrobotics_darknet_ros/lib/openrobotics_darknet_ros/detector_node: error while loading shared libraries: libdarknet.so: cannot open shared object file: No such file or directory` **when launching the detector node**. 
The reason for this is the install rule https://github.com/AlexeyAB/darknet/blob/d2c3788539bb73d3cfcdb7a1d5fa6b29ae1ec0e1/CMakeLists.txt#L549:L555 that installs the shared library to `./src/darknet/libdarknet.so` as well as `./build/darknet/libdarknet.so` instead of `./build/darknet/libdarknet.so` and `./install/darknet/libdarknet.so`. I did not want to create a pull request for it on darknet as I was afraid it would break something for somebody and/or make the pull request difficult to go through. One option would be to make anther pull request to darknet and add a different install rule guarded by a `if(ament_cmake_FOUND)`.
Maybe you have an idea on how to resolve this in an elegant manner. I just changed this line to a **scoped `set_target_properties(openrobotics_darknet_ros PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)`** in the latest commit.